src/v/cluster/controller_backend.cc

```diff
@@ -885,522 +690,491 @@ void controller_backend::housekeeping() {
+
+    if (partition) {
+        vassert(
+          partition->get_log_revision_id() == *expected_log_rev,
+          "[{}] unexpected partition log_revision: {} (expected {})",
+          ntp,
+          partition->get_log_revision_id(),
+          *expected_log_rev);
     }
-    /**
-     * Cancelling partition movement may only be executed before the update
-     * finished. We only remove partition replicas when processing
-     * `update_finished` delta. This is why partitions will never have to be
-     * created with cancel/abort type of deltas.
-     */
-    vassert(
-      type == partition_operation_type::update
-        || type == partition_operation_type::force_update,
-      "Invalid reconciliation loop state. Partition replicas should not be "
-      "removed before finishing update, ntp: {}, current operation: {}, "
-      "target_assignment: {}",
-      ntp,
-      type,
-      data.target_assignment);
-    /**
-     * We expect partition replica to exists on current broker/shard. Create
-     * partition. we relay on raft recovery to populate partition
-     * configuration.
-     */
-    auto ec = co_await create_partition(
-      ntp, data.target_assignment.group, command_rev, command_rev, {});
-    // wait for recovery, we will mark partition as updated in next
-    // controller backend reconciliation loop pass
-    if (!ec) {
-        co_return errc::waiting_for_recovery;
+    if (claim_it != _ntp_claims.end()) {
+        vassert(
+          claim_it->second.log_revision == *expected_log_rev,
+          "[{}] unexpected partition claim log_revision: {} (expected {})",
+          ntp,
+          claim_it->second.log_revision,
+          *expected_log_rev);
     }
-    co_return ec;
-}
 
-ss::future<std::error_code> controller_backend::reset_partition(
-  model::ntp ntp,
-  const partition_assignment& target_assignment,
-  const replicas_t& prev_replicas,
-  const replicas_revision_map& replica_revisions,
-  model::revision_id cmd_revision) {
-    vlog(
-      clusterlog.trace,
-      "[{}] processing partition reset command with target replicas: {}, "
-      "previous replica set: {}, replica_revisions map: {}",
-      ntp,
-      target_assignment.replicas,
-      prev_replicas,
-      replica_revisions);
+    // After this point the partition is expected to exist on this node.
 
-    auto partition = _partition_manager.local().get(ntp);
-
-    if (has_local_replicas(_self, target_assignment.replicas)) {
-        model::revision_id target_rev = replica_revisions.at(_self);
+    if (!expected_on_cur_shard) {
+        // Partition is expected to exist on some other shard. If we still own
+        // it, shut it down and make it available for cross-shard move.
 
         if (partition) {
-            if (partition->get_revision_id() >= target_rev) {
-                co_return errc::success;
+            rs.set_cur_operation(
+              last_cmd_revision, partition_operation_type::reset, assignment);
```

Comment:
As a generic partition creation/deletion operation. Probably `update` would fit better, but it could be a cancellation as well, or the update could be already finished and we won't be able to determine its type. Though we can just use generic "update" here. WDYT?

Reply:
maybe an `update` would better express that the partition is actually being changed

Reply:
After some thought I think reset is better. The operation that caused this could be update, (force_)cancel, update_finished, or in the future even set_disabled. I think reset is the best umbrella term if we are not going to add a lot of ifs calculating the precise operation type.